src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -271,15 +282,47 @@ public String getHooksRootUrl() {
     }
 
     @DataBoundSetter
-    public void setSecretToken(Secret token) {
```

Comment:
Hmm this is breaking binary compatibility and JCasC compatibility. I believe the proper way is to retain the getter and setter and make them use the credential.

Not sure if the `setSecretToken` should retain the `@DataBoundSetter` migration is always hard 😰